### PR TITLE
Refactor some test code

### PR DESCRIPTION
**Pull request by stuartskelton**
Hi all,

Here is a little refactor within the test suite.

You may be wondering why? A few months back I was messing about with AWS-Lamba, and it would have been nice to just have clear interfaces for the todolist storage layer/ Searching.

https://github.com/stuartskelton/todolist/commit/758f95a0400d21dfe83116d106284eee45e7157f

is a very rough WIP in abstracting the storage. In another branch I have the notion of a 'TodoCriteria' which breaks what the user is searching to help draw a lines between UI and search intent.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -6,8 +6,8 @@ import (
 	"strings"
 
 	"github.com/fatih/color"
-	"github.com/gammons/todolist/todolist"
 	"github.com/skratchdot/open-golang/open"
+	"github.com/stuartskelton/todolist/todolist"
```
Comment (stuartskelton) [inline main.go]: I will fix this if your happy for this to be merged,

File: todolist/todo_test.go [hunk 2]
```diff
@@ -21,3 +25,36 @@ func TestValidity(t *testing.T) {
 		t.Error("Invalid todo is being reported as valid")
 	}
 }
+
+//SetUpTestMemoryTodoList sets up a fixtures test todolist
+func SetUpTestMemoryTodoList() *TodoList {
+	store := &MemoryStore{}
```
Comment (stuartskelton) [inline todolist/todo_test.go]: These todos are the todos in the test file, I have moved them in here so its safe from meddling.

File: main.go [hunk 3]
```diff
@@ -41,102 +41,138 @@ func usage() {
 	yellow.Println("\ttodo a +work +verify did @john fix the build\\?")
 
 	blueBold.Println("\nListing todos")
-	fmt.Println("  When listing todos, you can filter and group the output.\n")
+	fmt.Println("  When listing todos, you can filter and group the output.")
+	fmt.Println()
```
Comment (stuartskelton) [inline main.go]: Go likes to complain if you have a "\n" on the end of a Println().

Comment (gammons) [reply main.go]: Yep, good fix!

File: todolist/file_store_test.go [hunk 4]
```diff
@@ -8,13 +8,18 @@ import (
...
 
-func TestSave(t *testing.T) {
-	store := &FileStore{FileLocation: "todos.json"}
-	todos, _ := store.Load()
-	store.Save(todos)
+	// reload the fixture data
+	store1 := &FileStore{FileLocation: testFilename}
+	todos, _ := store1.Load()
+	assert.Equal(todos[1].Subject, "this is the first subject", "")
+	assert.Equal(todos[0].Subject, "this is an non-fixture subject", "")
```
Comment (gammons) [inline todolist/file_store_test.go]: 2 comments about this test:

1.  I like tests formatted using [Thoughtbot's recommendations](https://robots.thoughtbot.com/how-we-test-rails-applications), so something along the format of:
```
setup

exercise

assertions
```

The main point is that the exercise has blank lines above and below it.  In this case, I think `store1.Load()` is probably the exercise.  The stuff above it is setup.

2. not sure if the comments are needed.

**Comment by gammons** [comment]
had one comment about the bigger test.  Otherwise, I think this PR looks great, and it's a nice cleanup from what it was!

**Pull request merged**